python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/emr_serverless.py

```diff
@@ -0,0 +1,289 @@
+                    context.log.warning(
+                        f"[pipes] Dagster process interrupted! Will terminate external {self.AWS_SERVICE_NAME} workload."
+                    )
+                    self._terminate(context, start_response)
+                raise
+
+    def _enrich_start_params(
+        self,
+        context: OpExecutionContext,
+        session: PipesSession,
+        params: "StartJobRunRequestRequestTypeDef",
+    ) -> "StartJobRunRequestRequestTypeDef":
+        # inject Dagster tags
+        tags = params.get("tags", {})
+        tags.update(
+            {
+                "dagster/run_id": context.run_id,
+            }
+        )
+        params["tags"] = tags
+
+        # inject env variables via --conf spark.executorEnv.env.<key>=<value>
+
+        dagster_env_vars = {}
+
+        if isinstance(self.context_injector, PipesEnvContextInjector):
+            dagster_env_vars.update(session.get_bootstrap_env_vars())
+
+        if "jobDriver" not in params:
+            params["jobDriver"] = {}
+
+        if "sparkSubmit" not in params["jobDriver"]:
+            params["jobDriver"]["sparkSubmit"] = {}
+
+        params["jobDriver"]["sparkSubmit"]["sparkSubmitParameters"] = params.get(
+            "jobDriver", {}
+        ).get("sparkSubmit", {}).get("sparkSubmitParameters", "") + "".join(
+            [
+                f" --conf spark.emr-serverless.driverEnv.{key}={value}"
+                for key, value in dagster_env_vars.items()
+            ]
+        )
+
+        return params
+
+    def _start(
+        self, context: OpExecutionContext, params: "StartJobRunRequestRequestTypeDef"
+    ) -> "StartJobRunResponseTypeDef":
+        response = self._client.start_job_run(**params)
+        job_run_id = response["jobRunId"]
+        context.log.info(
+            f"[pipes] {self.AWS_SERVICE_NAME} job started with job_run_id {job_run_id}"
+        )
+        return response
+
+    def _wait_for_completion(
+        self, context: OpExecutionContext, start_response: "StartJobRunResponseTypeDef"
+    ) -> "GetJobRunResponseTypeDef":
+        job_run_id = start_response["jobRunId"]
+
+        while response := self._client.get_job_run(
+            applicationId=start_response["applicationId"],
+            jobRunId=job_run_id,
+        ):
+            state: JobRunStateType = response["jobRun"]["state"]
+
+            if state in ["FAILED", "CANCELLED", "CANCELLING"]:
+                context.log.error(
+                    f"[pipes] {self.AWS_SERVICE_NAME} job {job_run_id} terminated with state: {state}. Details:\n{response['jobRun'].get('stateDetails')}"
+                )
+                raise RuntimeError(f"{self.AWS_SERVICE_NAME} job failed")
+
+            elif state == "SUCCESS":
+                context.log.info(
+                    f"[pipes] {self.AWS_SERVICE_NAME} job {job_run_id} completed with state: {state}"
+                )
+                return response
+
+            time.sleep(10)
```

Comment:
how did we select 10 seconds? Should this be configurable?

Reply:
added a parameter